src/main/java/javaslang/collection/Iterator.java

```diff
@@ -166,6 +167,16 @@ public T next() {
         };
     }
 
+    @Override
+    default Iterator<T> iterator() {
+        return this;
```

Comment:
I think not, because Iterators are mutable/not reusable. The new Iterator would wrap the other and be the identity operation. It is semantically the same as `this`.